File: university-service/src/main/java/com/university/service/AttendanceService.java
package com.university.service;

import com.university.dto.AttendanceReqDTO;
import com.university.dto.AttendanceRespDTO;
import java.util.List;

public interface AttendanceService {
    AttendanceRespDTO markAttendance(AttendanceReqDTO dto);
    AttendanceRespDTO updateAttendance(Long id, AttendanceReqDTO dto);
    AttendanceRespDTO getAttendanceById(Long id);
    List<AttendanceRespDTO> getAllAttendance();
    List<AttendanceRespDTO> getAttendanceByStudent(Long studentId);
    List<AttendanceRespDTO> getAttendanceBySubject(Long subjectId);
    List<AttendanceRespDTO> getAttendanceBySubjectAndSemester(Long subjectId, Long semesterId);
    void deleteAttendance(Long id);
}

File: university-frontend/university-frontend/src/services/attendanceService.js
import axiosInstance from '../api/axiosInstance';


// Fetch all attendance records
export const getAllAttendance = () => axiosInstance.get('/attendance');

// Fetch a specific attendancebystudent by ID
export const getAttendanceByStudent = (studentId) => axiosInstance.get(`/attendance/student/${studentId}`);

// Fetch a specific attendancebysubjectandsemester by ID
export const getAttendanceBySubjectAndSemester = (subjectId, semesterId) =>
  axiosInstance.get(`/attendance/subject/${subjectId}/semester/${semesterId}`);
export const markAttendance = (data) => axiosInstance.post('/attendance', data);

// Update an existing attendance record
export const updateAttendance = (id, data) => axiosInstance.put(`/attendance/${id}`, data);

// Delete a attendance record
export const deleteAttendance = (id) => axiosInstance.delete(`/attendance/${id}`);
